packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -474,122 +474,120 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
           ? updateVisit(visitToEdit?.uuid, payload, abortController)
           : saveVisit(payload, abortController)
         )
-          .pipe(first())
-          .subscribe({
-            next: (response) => {
-              if (config.showServiceQueueFields && queueLocation && service && priority) {
-                // retrieve values from the queue extension
-                setVisitUuid(response.data.uuid);
-
-                saveQueueEntry(
-                  response.data.uuid,
-                  service,
-                  patientUuid,
-                  priority,
-                  status,
-                  sortWeight,
-                  queueLocation,
-                  visitQueueNumberAttributeUuid,
-                  abortController,
-                ).then(
-                  ({ status }) => {
+          .then((response) => {
+            if (config.showServiceQueueFields && queueLocation && service && priority) {
+              // retrieve values from the queue extension
+              setVisitUuid(response.data.uuid);
+
+              saveQueueEntry(
+                response.data.uuid,
+                service,
+                patientUuid,
+                priority,
+                status,
+                sortWeight,
+                queueLocation,
+                visitQueueNumberAttributeUuid,
+                abortController,
+              ).then(
+                () => {
+                  mutateCurrentVisit();
+                  mutateVisits();
+                  mutateInfiniteVisits();
+                  mutateQueueEntry();
+                  showSnackbar({
+                    kind: 'success',
+                    title: t('visitStarted', 'Visit started'),
+                    subtitle: t('queueAddedSuccessfully', `Patient added to the queue successfully.`),
+                  });
+                },
+                (error) => {
+                  showSnackbar({
+                    title: t('queueEntryError', 'Error adding patient to the queue'),
+                    kind: 'error',
+                    isLowContrast: false,
+                    subtitle: error?.message,
+                  });
+                },
+              );
+
+              if (config.showUpcomingAppointments && upcomingAppointment) {
+                updateAppointmentStatus('CheckedIn', upcomingAppointment.uuid, abortController).then(
+                  () => {
                     mutateCurrentVisit();
                     mutateVisits();
                     mutateInfiniteVisits();
```

Comment:
Can we make these 3 mutates outside of the `if` conditions since they are related to the core visit form itself?

Reply:
@vasharma05 Let's move that to it's own ticket.